File: C/beginner/1051.c
// https://judge.beecrowd.com/pt/problems/view/1051
// Imposto de Renda

// Libraries
#include <stdio.h>

// Main Function
int main(){
    // Variables
    float salary, tax;
    // Input
    scanf("%f", &salary);
    // Process
    if(salary <= 2000.00){
        printf("Isento\n");
    }else if(salary <= 3000.00){
        tax = (salary - 2000.00) * 0.08;
        printf("R$ %.2f\n", tax);
    }else if(salary <= 4500.00){
        tax = (1000.00 * 0.08) + ((salary - 3000.00) * 0.18);
        printf("R$ %.2f\n", tax);
    } else{
        tax = (1000.00 * 0.08) + (1500.00 * 0.18) + ((salary - 4500.00) * 0.28);
        printf("R$ %.2f\n", tax);
    }

    // Ending code
    return 0;
}

File: Haskell/beginner/1051.hs
-- https://judge.beecrowd.com/pt/problems/view/1051
-- Imports
import Text.Printf

-- Main function
main = do
    -- Requesting salary value and assign into a variable named salary
    salary <- readLn :: IO Double

    -- Finding the amount of tax and presenting it
    case salary of
        salary | salary <= 2000.00 -> printf "Isento\n"
        salary | salary >= 2000.1 && salary <= 3000.00
            -> printf "R$ %.2f\n" ((salary - 2000.00) * 0.08)
        salary | salary >= 3000.01 && salary <= 4500.00
            -> printf "R$ %.2f\n" ((1000.00 * 0.08) + ((salary - 3000.00) * 0.18))
        _ -> printf "R$ %.2f\n" ((1000.00 * 0.08) + (1500.00 * 0.18) + ((salary - 4500.00) * 0.28))

-- By Fernando Serra
-- https://github.com/fernandosserra